### PR TITLE
Feature/symfony5

### PR DESCRIPTION
**Pull request by Chris53897**
hi, i made some changes. i hope you like it.

make the bundle work with the lastest symfony 5.x version and 6.x dev
autowiring for controllers
add github actions
add php 7.2 features
update composer.json
remove composer.lock (should not be commited for bundles)

### CODE REVIEW

File: .github/workflows/code_checks.yaml
Comment (MKodde) [inline .github/workflows/code_checks.yaml:39]: I'd love to see the other tests we run on Travis also being run on GH Actions. We utilize Ant for running tests, see the build.xml. I do not propose using Ant but maybe we can put some energy into moving the test set to the Composer scripts section?

File: .travis.yml
Comment (MKodde) [inline .travis.yml]: We had this configured so our production version would not be allowed to fail (PHP 7.2) But having it configured like this is fine for now, I see no reason why any php 7 version would start failing in the near feature.

On a side note. If we move away all code quality tests from Travis to Github Actions, we would be able to shut down Travis. Also see my previous comment for some suggestions.

File: CHANGELOG.md
Comment (MKodde) [inline CHANGELOG.md]: I suggest making this a new major release as support for older PHP versions is now fully dropped. So a 2.0.0 release is what I would expect here.

File: CHANGELOG.md
Comment (MKodde) [inline CHANGELOG.md]: Typo: ~an~ and

File: README.md
Comment (MKodde) [inline README.md:139]: :1st_place_medal:

File: composer.json
Comment (MKodde) [inline composer.json]: Perhaps add a <9.0-dev restriction. I feel this added defense will prevent some installation issues in the future.

File: phpunit.xml
Comment (MKodde) [inline phpunit.xml]: The previous multiline format of the `phpunit` element allowed a quick look into how PHPUnit is configured. Maybe reformat that element again like that?

File: src/Controller/HealthController.php
Comment (MKodde) [inline src/Controller/HealthController.php:31]: Why extend the `AbstractController`? We use none of it's features. But maybe this is a framework obligation introduced in a newer Symfony version than I'm acquainted with? If possible, please remove this extension. This also applies for the InfoController

File: CHANGELOG.md
Comment (MKodde) [inline CHANGELOG.md]: I would be happy releasing this in a single big bang version ~2.0.0~ 3.0.0.. But having smaller releases can also be a viable release strategy.

*Edit*, changed the targeted release number.

File: src/HealthCheck/DoctrineConnectionHealthCheck.php
Comment (MKodde) [inline src/HealthCheck/DoctrineConnectionHealthCheck.php:58]: :1st_place_medal: thanks for pinching the deprecation warning!

File: src/Resources/config/routing.yml
Comment (MKodde) [inline src/Resources/config/routing.yml:5]: I'm not the biggest fan of using the __invoke method over a named controller action. But alas, this works too!

**Review by MKodde — COMMENTED** [review]
Thanks for your effort @Chris53897! Love the work you did here. I have some questions and suggestions. But all in all, this looks very good indeed.

File: CHANGELOG.md
Comment (Chris53897) [reply CHANGELOG.md]: 2.1.0 is already released. Should i write 3.0 ?

File: src/Resources/config/routing.yml
Comment (Chris53897) [reply src/Resources/config/routing.yml:5]: I run into some trouble with removing the deprecation with the explicit method names. That was the easiest way that works. So i choose that solution

File: src/Controller/HealthController.php
Comment (Chris53897) [reply src/Controller/HealthController.php:31]: I changed the code to use the default json return $this->json(...) from the AbstractController.
In my eyes it is always good to use the default symfony features. But i can revert the json responsens and the extends AbstractController if you wish. Please let me know

File: phpunit.xml
Comment (Chris53897) [reply phpunit.xml]: Good idea. I changed it back to multilines. These one-liner is the default after i executed the migration option from phpunit.

File: composer.json
Comment (Chris53897) [reply composer.json]: i am not sure if this is correct. "php": ">=7.2, <9.0-dev"
Can you please check it

File: .github/workflows/code_checks.yaml
Comment (Chris53897) [reply .github/workflows/code_checks.yaml:39]: I have never used Ant or Travis. PHP-CS-Fixer from EasyAdminBundle was working fine.
I searched for your github actions. Found this one https://github.com/OpenConext/OpenConext-engineblock/blob/master/.github/workflows/test-integration.yml
Will try to integrate more quality checks

File: .travis.yml
Comment (Chris53897) [reply .travis.yml]: I removed the 3 files beloging to travis

File: .github/workflows/code_checks.yaml
Comment (Chris53897) [reply .github/workflows/code_checks.yaml:39]: I moved the 5 scripts from the build file to composer-script section. But i need to test if i did this correct.
Not that this is run after composer upgrade at the end-user.

Comment (Chris53897) [reply .github/workflows/code_checks.yaml:39]: Look like it works fine. 
a ) Checks are performed. [See Testrun](https://github.com/Chris53897/Monitor-bundle/runs/3780911967?check_suite_focus=true) under "Install dependencies"
b) Checks do not run if bundle is required in a project. See below.

But of course this can be changed to somethin else. Please let me know.

`statusboard composer update
Loading composer repositories with package information                                                                                                  Updating dependencies                                 
Lock file operations: 0 installs, 1 update, 0 removals
  - Upgrading openconext/monitor-bundle (dev-feature/symfony5 51896e2 => dev-feature/symfony5 417fb28)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 0 installs, 1 update, 0 removals
  - Downloading openconext/monitor-bundle (dev-feature/symfony5 417fb28)
  - Upgrading openconext/monitor-bundle (dev-feature/symfony5 51896e2 => dev-feature/symfony5 417fb28): Extracting archive
Generating autoload files
composer/package-versions-deprecated: Generating version class...
composer/package-versions-deprecated: ...done generating version class
94 packages you are using are looking for funding.
Use the `composer fund` command to find out more!

What about running composer global require symfony/thanks && composer thanks now?
This will spread some 💖  by sending a ★  to the GitHub repositories of your fellow package maintainers.

Run composer recipes at any time to see the status of your Symfony recipes.

Executing script cache:clear [OK]
Executing script assets:install --symlink --relative public [OK]
`

File: src/Controller/HealthController.php
Comment (Chris53897) [reply src/Controller/HealthController.php:31]: They removed this code in the service.yaml (after i upgraded the recipes in an other project to symfony 5.4.dev)
I guess it is still possible to manuel tag Services as controllers. But extend from AbstractController is the safer way.

`# controllers are imported separately to make sure services can be injected
  # as action arguments even if you don't extend any base controller class
    App\Controller\:
        resource: '../src/Controller/'
        tags: ['controller.service_arguments']
`

**Comment by Chris53897** [comment]
Did i complete all of your suggestions? Or is there anything else i can do to help for this PR to get merged?

File: CHANGELOG.md
Comment (MKodde) [reply CHANGELOG.md]: Yes, I noticed this later. Going for 3.0.0 seems the logical release :+1:

File: composer.json
Comment (MKodde) [reply composer.json]: My Composer had no issue with `"php": ">=5.4,<9.0-dev" Did you run into issues while adding that additional constraint?

File: src/Controller/HealthController.php
Comment (MKodde) [reply src/Controller/HealthController.php:31]: Personally I prefer to couple as loosely to Symfony as possible. Using the `json` feature provided by the abstract controller is a kind of legitimate argument to use the abstract controller. Feel free to leave this as is!

File: composer.json
Comment (MKodde) [inline composer.json:45]: Using the `post-update-cmd` hook is a cool solution and a honest way to run these tests on Github Actions. 

But for running the tests outside of a composer install/update context, it might also be usefull to expose the set of scripts under a `tests` or other fitting script name.

File: .github/workflows/code_checks.yaml
Comment (MKodde) [inline .github/workflows/code_checks.yaml:4]: I don't see the workflow running on this PR. Looking at our EngineBlock test-integration workflow:

```
on:
    push:
        branches:
            - master
    pull_request:
```

Perhaps you need the target on the master branch to trigger starting the builds?

**Review by MKodde — COMMENTED** [review]
I think you are getting real close to having your PR merged, I found a couple of things that might need some attention.

Also, if you feel like it. Would you be so kind to look at those points? And feel free to discuss my opinions.

File: composer.json
Comment (Chris53897) [reply composer.json:45]: I followed you suggestion and added a **tests** section.

File: .github/workflows/code_checks.yaml
Comment (Chris53897) [reply .github/workflows/code_checks.yaml:4]: this PR targets **master**
But this Repo has no github action in master. 

> Looking at our EngineBlock test-integration workflow:
Maybe you are looking in the wrong repo?

So only Travis should be started `https://github.com/OpenConext/Monitor-bundle/blob/master/.travis.yml`

I do not know travis much. But i guess it is not triggered for PR.

```
branches:
  only:
    - master
    - develop
```

You can see [my GitHub Actions Testruns](https://github.com/Chris53897/Monitor-bundle/actions)

If you want you can manually add these 2 files to your master-branch https://github.com/Chris53897/Monitor-bundle/tree/feature/symfony5/.github/workflows
After that, all new PR (runs) should activate the github actions.
At forks, you have to manually enable Testruns, but i guess this should not be necessary on the orginal.

Comment (Chris53897) [reply .github/workflows/code_checks.yaml:4]: The CI-Runs are from the target-Branch or master. Not from the PR

**Comment by Chris53897** [comment]
I added another github action test for upcoming symfony versions. That will fail for some time, until conflicts are resolved at external libaries. You can toogle these 2 Lines to make an precheck. (In this case not every symfony package is restricted to 6.0.dev) Restricting packages listed in "symfony/symfony" to "6.0.x@dev"

env:
          #SYMFONY_REQUIRE: '6.0.x@dev'

**Comment by Chris53897** [comment]
Please make an manual test if everythinks works correctly for you before merging.
CI  test is only as good, as the Test-Coverage.

I have testet it right know in my demo app with Symfony 5.4.0-dev and it works.

**Review by MKodde — APPROVED** [review]

**Comment by MKodde** [comment]
Sorry for keeping this in the fridge for so long. Thanks for all you efforts.

Merging!

**Pull request merged**
